DSESwift/DataStorageEngine.swift

```diff
@@ -70,4 +165,121 @@ public class DataStorageEngine{
     public func getSources() -> [String]{
         return DatabaseHandler.getSources()
     }
+    
+    /**
+     * Returns enum DSEStatus indicating status of DSE.
+     * @return The DSEStatus, this could be either AWAITING_CREDENTIALS, AWAITING_SENSOR_PROFILES, READY.
+     **/
+    public func getStatus() -> DSEStatus{
+        if(self.config.sessionId == nil) {
```

Comment:
can the case occur where the sessionID is set but the appkey and userid are missing?